addons/auth_totp/controllers/home.py

```diff
@@ -58,7 +59,7 @@ def web_totp(self, redirect=None, **kwargs):
                     if request.geoip.city.name:
                         name += f" ({request.geoip.city.name}, {request.geoip.country_name})"
 
-                    key = request.env['auth_totp.device']._generate("browser", name)
+                    key = request.env['auth_totp.device'].sudo()._generate("browser", name, TRUSTED_DEVICE_APIKEY_AGE)
```

Comment:
Why there is a sudden need of `sudo()` on `_generate` ?
Is this something you addeed that required the addition of `sudo`? Then, should you put the `sudo` at the right place within the method ?